### PR TITLE
Fix and complete decorator API doc

### PR DESCRIPTION
**Pull request by jbogarthyde**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

Fix existing doc to be in correct places to be rendered
Add top-level descriptions and property descriptions where missing.
Edit and correct descriptions.

- [x] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] angular.io application / infrastructure changes
- [ ] Other... Please describe:

## What is the current behavior?

Issue Number: https://github.com/angular/angular/issues/28710

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: packages/core/src/metadata/directives.ts [hunk 1]
```diff
@@ -622,23 +637,49 @@ export const Pipe: PipeDecorator = makeDecorator(
...
+  *   normalizedBankName: string;
+  * }
+  *
+  * @Component({
+  *   selector: 'app',
+  *   template: `
+  *     <bank-account bankName="RBC" account-id="4747"></bank-account>
+  *   `
+  * })
+  *
```
Comment (brandonroberts) [inline packages/core/src/metadata/directives.ts]: Remove extra line here

File: packages/core/src/metadata/directives.ts [hunk 2]
```diff
@@ -841,6 +849,10 @@ export const HostBinding: HostBindingDecorator =
  * @publicApi
  */
 export interface HostListenerDecorator {
+  /**
+   * Decorator that declares a CSS event to listen for,
```
Comment (brandonroberts) [inline packages/core/src/metadata/directives.ts]: I believe references to CSS event(s) should be DOM event(s) to be more correct. 

For reference: https://developer.mozilla.org/en-US/docs/Web/Events

File: packages/core/src/metadata/directives.ts [hunk 3]
```diff
@@ -715,7 +714,7 @@ const initializeBaseDef = (target: any): void => {
 };
 
 /**
- * Does the work of creating the `ngBaseDef` property for the @Input and @Output decorators.
+ * Does the work of creating the `ngBaseDef` property for the `@Input` and `@Output` decorators.
```
Comment (brandonroberts) [inline packages/core/src/metadata/directives.ts]: Remove the `@` signs?

File: packages/core/src/metadata/directives.ts [hunk 4]
```diff
@@ -572,7 +572,22 @@ export const Component: ComponentDecorator = makeDecorator(
...
+   *
+   * ```
+   * {{ exp | myPipe }}
+   * ```
+   *
+   * The result of the expression is passed to the pipe's `transform()` method.
+   *
+   * A pipe must belong to an NgModule in order for it to be available
+   * to a template. To make it a member of an NgModule,
+   * list it in the `declarations` field of the `@NgModule` metadata.
```
Comment (brandonroberts) [inline packages/core/src/metadata/directives.ts]: We don't usually use the `@` sign. The link doesn't include it either. Remove?

File: packages/core/src/di/metadata.ts [hunk 5]
```diff
@@ -237,20 +237,18 @@ export const Host: HostDecorator = makeParamDecorator('Host');
...
-   * ### Example
    *
    * Suppose we have an `<input>` element and want to know its `type`.
    *
    * ```html
    * <input type="text">
    * ```
    *
-   * A decorator can inject string literal `text` like so:
+   * The following example uses the decorator to inject the string literal `text`.
```
Comment (brandonroberts) [inline packages/core/src/di/metadata.ts]: Remove the ES5 annotation example below

**Review by brandonroberts — CHANGES_REQUESTED** [review]

**Review by mhevery — APPROVED** [review]

**Review by brandonroberts — APPROVED** [review]

**Comment by brandonroberts** [comment]
Caretaker note: This is a docs only change and should not impact g3. Please run presubmit.

**Comment by AndrewKushnir** [comment]
[Presubmit](http://test/OCL:236679182:BASE:236679245:1551724265250:7c20c0c6)

**Comment by AndrewKushnir** [comment]
@jbogarthyde, @brandonroberts there was a conflict while merging this PR into patch branch, so I've merged it to master only. Could you please create a version of this PR against patch (7.2.x) branch? Thank you.

**Pull request closed without merging**